cli/attach.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/alibaba/pouch/pkg/ioutils"
+	"github.com/alibaba/pouch/pkg/term"
+)
+
+// AttachDescription is used to describe attach command in detail and auto generate command doc.
+var AttachDescription = "Attach local standard input, output, and error streams to a running container"
+
+// AttachCommand is used to implement 'attach' command.
+type AttachCommand struct {
+	baseCommand
+
+	// flags for attach command
+	noStdin    bool
+	detachKeys string
+}
+
+// Init initialize "attach" command.
+func (ac *AttachCommand) Init(c *Cli) {
+	ac.cli = c
+	ac.cmd = &cobra.Command{
+		Use:   "attach [OPTIONS] CONTAINER",
+		Short: "Attach local standard input, output, and error streams to a running container",
+		Long:  AttachDescription,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return ac.runAttach(args)
+		},
+		Example: ac.example(),
+	}
+	ac.addFlags()
+}
+
+// addFlags adds flags for specific command.
+func (ac *AttachCommand) addFlags() {
+	flagSet := ac.cmd.Flags()
+	flagSet.BoolVar(&ac.noStdin, "no-stdin", false, "Do not attach STDIN")
+	flagSet.StringVar(&ac.detachKeys, "detach-keys", "", "Override the key sequence for detaching a container")
+}
+
+// runAttach is used to attach a container.
+func (ac *AttachCommand) runAttach(args []string) error {
+	name := args[0]
+
+	ctx := context.Background()
+	apiClient := ac.cli.Client()
+
+	in, out, err := setRawMode(!ac.noStdin, false)
```

Comment:
only terminal can support rawMode. we need to detect the command is running in the terminal.

Reply:
Done